server/datastore/mysql/software_installers.go

```diff
@@ -70,6 +70,24 @@ func (ds *Datastore) GetSoftwareInstallDetails(ctx context.Context, executionId
 		}
 		return nil, ctxerr.Wrap(ctx, err, "get software install details")
 	}
+
+	expandedInstallScript, err := ds.ExpandEmbeddedSecrets(ctx, result.InstallScript)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "expanding secrets in install script")
+	}
+	expandedPostInstallScript, err := ds.ExpandEmbeddedSecrets(ctx, result.PostInstallScript)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "expanding secrets in post-install script")
+	}
+	expandedUninstallScript, err := ds.ExpandEmbeddedSecrets(ctx, result.UninstallScript)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "expanding secrets in uninstall script")
+	}
+
+	result.InstallScript = expandedInstallScript
+	result.PostInstallScript = expandedPostInstallScript
+	result.UninstallScript = expandedUninstallScript
+
```

Comment:
This looks like for the UI/activity. We should not display secrets in the UI -- leave them hidden.

The secrets should only be expanded for the device/host.

Reply:
This method is actually only used when orbit hits its endpoint to get software installer information, we provide the scripts in one endpoint https://github.com/fleetdm/fleet/blob/secrets-software-scripts/server/service/orbit.go#L1183

Reply:
It's not used by the UI

Reply:
Ok. Would be nice to rename the method to flag that it exposes secrets, but we can merge as is.

Reply:
I agree it looks like it would though, it's not the best name